pkg/reconciler/build/build.go

```diff
@@ -153,6 +153,36 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return nil
 	}
 
+	// If the build's status is cancelled, kill resources and update status
+	if isCancelled(build.Spec) {
+		logger.Warnf("Build has been cancelled: %v", build.Name)
+		build.Status.SetCondition(&duckv1alpha1.Condition{
+			Type:    v1alpha1.BuildSucceeded,
+			Status:  corev1.ConditionFalse,
+			Reason:  "BuildCancelled",
+			Message: fmt.Sprintf("Build %q was cancelled", build.Name),
+		})
+		if err := c.updateStatus(build); err != nil {
+			return err
+		}
+		if build.Status.Cluster == nil {
+			logger.Warnf("build %q has no pod running yet", build.Name)
+			return nil
+		}
+		p, err := c.kubeclientset.CoreV1().Pods(build.Namespace).Get(build.Status.Cluster.PodName, metav1.GetOptions{})
```

Comment:
I think I already mentioned this(?) in previous comment about reusing `build.Status.Cluster.PodName` for Delete API directly. Sorry to repeat it again

Reply:
oh right, didn't update that :sweat: